runtime/oti/j9nonbuilder.h

```diff
@@ -383,6 +383,16 @@ typedef struct J9JFRExecutionSample {
 
 #define J9JFREXECUTIONSAMPLE_STACKTRACE(jfrEvent) ((UDATA*)(((J9JFRExecutionSample*)(jfrEvent)) + 1))
 
+typedef struct J9JFRJVMInformation {
+    struct J9UTF8 *jvmName;
```

Comment:
I should have mentioned this earlier. You can add entries to the string table via `addStringUTF8Entry` and just store a U_32 index in the struct.

Also, this struct should be defined in JFRConstantPoolTypes.hpp, you can call it `JVMInformationEntry`

Reply:
Actually I was trying to do that. I was using `writeLEB128(stringIndex)` instead of `writeUTF8String` but I was getting something like "invalid string encoding" when reading the jfr file. What am I missing here?